tests/rustdoc/inline_cross/issue-76736-2.rs

```diff
@@ -1,16 +1,18 @@
 //@ aux-build:issue-76736-1.rs
 //@ aux-build:issue-76736-2.rs
 
+// https://github.com/rust-lang/rust/issues/124635
+
 #![crate_name = "foo"]
 #![feature(rustc_private)]
 
 extern crate issue_76736_1;
 extern crate issue_76736_2;
 
 // @has foo/struct.Foo.html
-// @has - '//*[@class="impl"]//h3[@class="code-header"]' 'MaybeResult'
+// @!has - '//*[@class="impl"]//h3[@class="code-header"]' 'MaybeResult'
```

Comment:
I don't know, couldn't this impl be useful? The user is explicitly opting into using internal APIs with `#![feature(rustc_private)]`, then why should we refuse them access to potential useful information (here: “`impl MaybeResult for Foo`” via the upstream blanket impl, which they can indeed make use of)

Reply:
Well, the std library doesn't contain `#[feature(rustc_internal)]` directly as you probably know but yeah.

There are other users of `rustc_internal`. I' on mobile rn but yesterday I grepped through `compiler/` and saw the feature being enabled for one of the alternate backends (cranelift or gcc, I don't remember) and I assume `rustfmt` enables it too for `rustc_{parse,ast}`?

I'm just nitpicking but I don't wanna block